s3auth-relay/src/main/java/com/s3auth/relay/FtpFacade.java

```diff
@@ -0,0 +1,248 @@
+        Executors.newScheduledThreadPool(2, new VerboseThreads("FTP-front"));
+
+    /**
+     * Executor service, with consuming threads.
+     */
+    private final transient ScheduledExecutorService backend =
+        Executors.newScheduledThreadPool(
+            FtpFacade.THREADS,
+            new VerboseThreads("FTP-back")
+        );
+
+    /**
+     * Blocking queue of ready-to-be-processed sockets.
+     */
+    private final transient BlockingQueue<Socket> sockets =
+        new SynchronousQueue<Socket>();
+
+    /**
+     * Server socket.
+     */
+    private final transient ServerSocket server;
+
+    /**
+     * Public ctor.
+     * @param hosts Hosts
+     * @param port Port number
+     * @throws IOException If can't initialize
+     */
+    FtpFacade(@NotNull final Hosts hosts, final int port)
+        throws IOException {
+        this.server = new ServerSocket(port);
+        final FtpThread thread = new FtpThread(this.sockets, hosts);
+        final Runnable runnable = new VerboseRunnable(
+            new FtpFacade.FTPThreadRunnable(thread), true, true
+        );
+        for (int idx = 0; idx < FtpFacade.THREADS; ++idx) {
+            this.backend.scheduleWithFixedDelay(
+                runnable,
+                0L, 1L, TimeUnit.NANOSECONDS
+            );
+        }
+    }
+
+    /**
+     * Start listening to the ports.
+     */
+    public void listen() {
+        this.frontend.scheduleWithFixedDelay(
+            new VerboseRunnable(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        FtpFacade.this.process(FtpFacade.this.server);
+                    }
+                }
+            ),
+            0L, 1L, TimeUnit.NANOSECONDS
+        );
+    }
+
+    @Override
+    public void close() throws IOException {
+        try {
+            this.shutdown(this.frontend);
+            this.shutdown(this.backend);
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            throw new IOException(ex);
+        }
+        this.server.close();
+    }
+
+    /**
+     * Process one server socket.
+     * @param svr The server socket
+     */
+    private void process(final ServerSocket svr) {
+        final Socket socket;
+        try {
+            socket = svr.accept();
+        } catch (final IOException ex) {
+            throw new IllegalStateException(ex);
+        }
+        try {
+            if (!this.sockets.offer(socket, Tv.TEN, TimeUnit.SECONDS)) {
+                FtpFacade.overflow(socket);
+                Logger.warn(this, "too many open connections");
+            }
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    /**
+     * Report overflow problem to the socket and close it.
+     * @param socket The socket to report to
+     * @todo 213 implement
```

Comment:
This one looks implemented, so todo could be removed, if not, then please add `#` and some more information how to implement it